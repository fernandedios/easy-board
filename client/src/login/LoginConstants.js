export const DO_LOGIN = 'doLogin';
