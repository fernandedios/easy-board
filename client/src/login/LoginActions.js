import axios from 'axios';
import { DO_LOGIN } from './LoginConstants';

export const doLogin = ({ email, password }) => async dispatch => {
    const query = `mutation {
        login(email: "${email}", password: "${password}")
    }`;
    const res = await axios.post(`${process.env.REACT_APP_API}/api`, { query });
    dispatch({ type: DO_LOGIN, payload: { ...res.data }});
};
