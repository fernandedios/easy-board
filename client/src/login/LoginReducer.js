import { DO_LOGIN } from './LoginConstants';

const loginReducer = function(state = null, action) {
    switch (action.type) {
        case DO_LOGIN:
            return action.payload || false;

        default:
            return state;
    }
};

export default loginReducer;
