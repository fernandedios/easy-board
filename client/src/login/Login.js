import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';

import * as actions from './LoginActions';
import FormValidator from '../utils/FormValidator';
import TextInput from  '../common/TextInput';
import Button from '../common/Button';

class Login extends Component {
    constructor(props) {
        super(props);

        this.submitted = false;
        this.validator = new FormValidator([
            {
                field: 'email',
                method: 'isEmpty',
                validWhen: false,
                message: 'Email address is required.'
            },
            {
                field: 'email',
                method: 'isEmail',
                validWhen: true,
                message: 'Please provide a valid email address'
            },
            {
                field: 'password',
                method: 'isEmpty',
                validWhen: false,
                message: 'Password is required'
            }
        ]);

        this.state = {
            email: '',
            password: '',
            validation: this.validator.valid()
        };

        this.handleEvent = this.handleEvent.bind(this);
        this.onSubmit = this.onSubmit.bind(this);
    }

    handleEvent(e) {
        e.preventDefault();
        this.setState({ [e.target.name]: e.target.value });
    }

    onSubmit(e) {
        e.preventDefault();

        const validation = this.validator.validate(this.state);
        this.setState({ validation });
        this.submitted = true;

        if (validation.isValid) {
            const { email, password } = this.state;
            this.props.doLogin({ email, password });
        }
    }

    render() {
        let validation = this.submitted ? this.validator.validate(this.state) : this.state.validation;

        return (
            <section id="login-box" className="form-box hero">
                <div className="hero-body">
                    <div className="container has-text-centered">
                        <div className="column is-6 is-offset-3">
                            <h3>Login</h3>
                            <p>Kindly login to continue.</p>
                            <div className="box">
                                <form onSubmit={this.onSubmit}>
                                    <TextInput message={validation.email.message} onChange={this.handleEvent} value={this.state.email} name="email" type="email" placeholder="Email Address" />
                                    <TextInput message={validation.password.message} onChange={this.handleEvent} value={this.state.password} name="password" type="password" placeholder="Password" />
                                    <Button type="submit" name="login-btn" classes="is-block is-info is-fullwidth" label="Login" />
                                </form>
                            </div>
                            <p className="has-text-grey">Don't have account? <Link to="/signup">Sign up here</Link></p>
                        </div>
                    </div>
                </div>
            </section>
        );
    }
}

const mapStateToProps = ({ auth }) => {
    return { auth }
};

export default connect(mapStateToProps, actions)(Login);
