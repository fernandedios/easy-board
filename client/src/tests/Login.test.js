import React from 'react';
import { shallow, mount } from 'enzyme';
import ReactTestRenderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { Login } from '../login/Login';

describe('<LoginForm />', () => {
    it('should be defined', () => {
        expect(Login).toBeDefined();
    });

    it('should have correct classNames', () => {
        const output = shallow(<Login />);
        expect(output.find('section').hasClass('form-box')).toEqual(true);
        expect(output.find('section').hasClass('hero')).toEqual(true);
    });

    it('should render input fields correctly', () => {
        const output = mount(
            <MemoryRouter><Login /></MemoryRouter>
        );

        expect(output.find('#email').length).toEqual(1);
        expect(output.find('#password').length).toEqual(1);
    });

    it('should render correctly', () => {
        const output = ReactTestRenderer.create(<MemoryRouter><Login /></MemoryRouter>).toJSON();
        expect(output).toMatchSnapshot();
    });
});
