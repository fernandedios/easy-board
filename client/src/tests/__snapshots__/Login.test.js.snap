// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> should render correctly 1`] = `
<section
  className="form-box hero"
  id="login-box"
>
  <div
    className="hero-body"
  >
    <div
      className="container has-text-centered"
    >
      <div
        className="column is-6 is-offset-3"
      >
        <h3>
          Login
        </h3>
        <p>
          Kindly login to continue.
        </p>
        <div
          className="box"
        >
          <form
            onSubmit={[Function]}
          >
            <div
              className="field"
            >
              <label
                className="label"
              />
              <div
                className="control has-icons-right"
              >
                <input
                  className="input"
                  id="email"
                  name="email"
                  onChange={[Function]}
                  placeholder="Email Address"
                  type="email"
                  value=""
                />
              </div>
              <p
                className="help is-danger"
              >
                
              </p>
            </div>
            <div
              className="field"
            >
              <label
                className="label"
              />
              <div
                className="control has-icons-right"
              >
                <input
                  className="input"
                  id="password"
                  name="password"
                  onChange={[Function]}
                  placeholder="Password"
                  type="password"
                  value=""
                />
              </div>
              <p
                className="help is-danger"
              >
                
              </p>
            </div>
            <button
              className="button is-block is-info is-fullwidth"
              id="login-btn"
              name="login-btn"
              type="submit"
            >
              Login
            </button>
          </form>
        </div>
        <p
          className="has-text-grey"
        >
          Don't have account? 
          <a
            href="/signup"
            onClick={[Function]}
          >
            Sign up here
          </a>
        </p>
      </div>
    </div>
  </div>
</section>
`;
